syft/frameworks/torch/tensors/interpreters/precision.py

```diff
@@ -38,10 +40,30 @@ def __init__(
         """
         super().__init__(id=id, owner=owner, tags=tags, description=description)
 
-        self.field = field
         self.base = base
         self.precision_fractional = precision_fractional
         self.kappa = kappa
+        if dtype == "long":
+            self.dtype = "long"
+            self.field = 2 ** 64
+        elif dtype == "int":
+            self.dtype = "int"
+            self.field = 2 ** 32
+        else:
+            # Since n mod 0 is not defined
+            warnings.warn("Prefer to use dtype instead of field")
+            if type(field) == int and field > 0:
+                if field <= 2 ** 32:
+                    self.dtype = "int"
+                    self.field = 2 ** 32
+                else:
+                    self.dtype = "long"
+                    self.field = 2 ** 64
+            else:
+                # Invalid args dtype and field
+                raise ValueError(
+                    "Unsupported arg value for dtype. Use dtype='long' or dtype='int'."
+                )
```

Comment:
Just a question: why can we have a custom field for AST but not for FPT?

Reply:
Initial plan was to not provide an option for custom field at all for both FPT and AST. But there are certain use cases in securenn that demand field to be 67 and 2**64-1(for dtype long) and 2**32-1(for dtype int). 
So had to introduce custom dtype option that should only be used internally. But we don't need anything of the sort for FPT.